### PR TITLE
[Android] feedback widgets tweak functions

### PR DESCRIPTION
**Pull request by arifBurakDemiray**

### CODE REVIEW

File: android/next.md [hunk 1]
```diff
@@ -1919,10 +1919,51 @@ config.setStarRatingDisableAskingForEachAppVersion(false);</code></pre>
...
+Countly.sharedInstance().feedback().presentRating(this);
+// or
+Countly.sharedInstance().feedback().presentRating(this, "tag_published");
+</code></pre>
+<h3 id="h_01J9TZ1VZSG732EYVH1X9M2D04">Advanced Feedback Widget</h3>
+<p>
+  You can access their related information as a list corresponding to the current
+  user's device ID, by providing a callback to the getAvailableFeedbackWidgets
+  method, which returns the list as the first parameter and error as the second:
 </p>
```
Comment (turtledreams) [inline android/next.md:1967]: we want to get rid of the getavailablewidgets and such below, thats why we introduce these methods

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: android/next.md (hunk 1 above)
Comment (arifBurakDemiray) [reply android/next.md:1967]: yes but are not we are going to keep the old doc for now?

File: android/next.md [hunk 2]
```diff
@@ -1919,51 +1919,95 @@ config.setStarRatingDisableAskingForEachAppVersion(false);</code></pre>
   </p>
 </div>
 <p>
-  After you have created widgets at your dashboard you can reach their related
-  information as a list, corresponding to the current user's device ID, by providing
-  a callback to the getAvailableFeedbackWidgets method, which returns the list
-  as the first parameter and error as the second:
+  After you have created widgets on your dashboard, you can present them using
+  the function mentioned below. If you would like to access widget data and available
+  widgets, you can read the <a href="#h_01J9TZ1VZSG732EYVH1X9M2D04">advanced</a>
```
Comment (AliRKat) [inline android/next.md:1924]: there was something wrong with this link you might wanna check it again. It doesn't go to the link or section when I click on it. I'd suggest adding the full link there, also maybe labeling "advanced section" as the link can be a better look

File: android/next.md [hunk 3]
```diff
@@ -1919,51 +1919,95 @@ config.setStarRatingDisableAskingForEachAppVersion(false);</code></pre>
...
+<p>
+  If an empty nameIDorTag is provided it will show the first feedback widget by
+  its type
+</p>
+<p>
+  The "devCallback" parameter has two callbacks: - "onClosed" which will be called
+  when the feedback widget is closed - "onFinished" which will be called on some
+  internal errors and it will direct the error via "error" parameter.
+</p>
+<h3 id="h_01J9V00SX40115R37EBQKDF1HJ">NPS</h3>
```
Comment (AliRKat) [inline android/next.md:1943]: Adding a few lines on top of the code blocks can make it look much better, I think. This way, it looks a bit alien to me.

File: android/next.md [hunk 4]
```diff
@@ -1919,51 +1919,95 @@ config.setStarRatingDisableAskingForEachAppVersion(false);</code></pre>
...
+  @Override public void onFinished(String error) {
+    if (error != null) {
+      Toast.makeText(ActivityExampleFeedback.this, "Encountered error while presenting the feedback widget: [" + error + "]", Toast.LENGTH_LONG).show();
+    }
   }
 });
 </code></pre>
-<h3 id="h_01HAVQDM5V90VKV6QA45CK8Z49">Manual Reporting</h3>
+<h3 id="h_01J9V00SX4HKSKS4X4ENXKK0RS">Survey</h3>
+<pre><code>Countly.sharedInstance().feedback().presentSurvey(Context context)
```
Comment (AliRKat) [inline android/next.md:1964]: Same with NPS

File: android/next.md [hunk 5]
```diff
@@ -1919,51 +1919,95 @@ config.setStarRatingDisableAskingForEachAppVersion(false);</code></pre>
...
+
+  @Override public void onFinished(String error) {
+    if (error != null) {
+      Toast.makeText(ActivityExampleFeedback.this, "Encountered error while presenting the feedback widget: [" + error + "]", Toast.LENGTH_LONG).show();
+    }
+  }
+});
+</code></pre>
+<h3 id="h_01J9V00SX48CVFHF7JMTM3E3VH">Rating</h3>
+<pre><code>Countly.sharedInstance().feedback().presentRating(Context context)
```
Comment (AliRKat) [inline android/next.md:1985]: Same with NPS and Survey :D

File: android/next.md [hunk 6]
```diff
@@ -1919,51 +1919,95 @@ config.setStarRatingDisableAskingForEachAppVersion(false);</code></pre>
...
+    if (error != null) {
+      Toast.makeText(ActivityExampleFeedback.this, "Encountered error while presenting the feedback widget: [" + error + "]", Toast.LENGTH_LONG).show();
+    }
+  }
+});
+</code></pre>
+<p>
+  For more in-depth information on retrieving feedback widgets, understanding object
+  structures, or presenting them yourself, please refer to the following
+  <a href="https://support.count.ly/hc/en-us/articles/9290669873305-A-deeper-look-at-SDK-concepts#h_01JA7E8H963VYQV6BNR6TB9489" target="_blank" rel="noopener">resource</a>.
```
Comment (AliRKat) [inline android/next.md:2008]: link didn't go to the section, is it me or is there a problem

**Review by AliRKat — CHANGES_REQUESTED** [review]
great work bro, i think with few lines this is ready to go

File: android/next.md (hunk 6 above)
Comment (arifBurakDemiray) [reply android/next.md:2008]: it is the new deeper look doc, this https://github.com/Countly/countly-sdk-documentation/pull/703

File: android/next.md (hunk 2 above)
Comment (arifBurakDemiray) [reply android/next.md]: lets remove from here, we mention it below

**Comment by turtledreams** [comment]
made it shorter:
image

**Review by turtledreams — APPROVED** [review]

**Review by AliRKat — APPROVED** [review]

**Comment by arifBurakDemiray** [comment]
But is not this become not that much descriptive, I loved the shorter one but it became less descriptive. Anyone should understand the way it is now but they need to think

**Comment by turtledreams** [comment]
> Member

maybe bold titled in between?

**Comment by arifBurakDemiray** [comment]
No I guess no need, If we think that they will not read the texts and they just do copy paste the code blocks, this is so enough. In a moment, I thought mb this is not complete, but from the eyes of the developer who will read this, I would do only copy pasta 😅

**Pull request merged**